migration-guide.adoc

```diff
@@ -58,6 +58,47 @@ Multiple component mappings for the same java class with different property mapp
 
 Passing null or not is now triggered by whether setting the parameter was called at all.  In other ords a distinction is made between calling `setParameter` passing null versus not calling `setParameter` at all.  In the first case, we pass along the null value; in the second we do not.
 
+=== Query
+
+=== HQL results
+
+HQL queries that use joins without specifying a select clause e.g. `from Person p join p.address` do not return anymore a `List<Object[]>`with an entry per join anymore, but a list of `Person`.
+
+The HQL query `select p,a from Person p join p.address a` returns instead a `List<Object[]>`.
+
+e.g.
+```
+@Entity
+class Person {
+    ...
+
+    @ManyToOne
+    Address address
+
+    ...
+}
+
+@Entity
+class Address {
+
+    ...
+
+}
+
+List<Person> result = session.createQuery("from Person p join p.address").list();
+
+List<Object[]> results = session.createQuery("select p, a from Person p join p.address a").list();
+
+```
+
+==== Stream
+
+`jakarta.persistence.Query#getResultStream()` and `org.hibernate.query.Query#stream()` do not return a `Stream` decorator anymore, so in order to close the underlying JDK `Stream` is now necessary to explicitly call the `Stream#close()` method. The JDK `Stream` documentation is quite explicit about the need for an explicit call to `close` by the user to avoid resource leakages, so we build upon this requirement.
```

Comment:
```suggestion
`jakarta.persistence.Query#getResultStream()` and `org.hibernate.query.Query#stream()` do not return a `Stream` decorator anymore, so in order to close the underlying IO resources, it is now necessary to explicitly call the `Stream#close()` method. The JDK `Stream` documentation is quite explicit about the need for an explicit call to `close` by the user to avoid resource leakages, so we build upon this requirement.
```

Reply:
:+1: